src/ssl.c

```diff
@@ -292,7 +292,11 @@ static int ticket_key_callback(unsigned char *key_name, unsigned char *iv, EVP_C
     Found:
         EVP_DecryptInit_ex(ctx, ticket->cipher.cipher, NULL, ticket->cipher.key, iv);
         HMAC_Init_ex(hctx, ticket->hmac.key, EVP_MD_block_size(ticket->hmac.md), ticket->hmac.md, NULL);
-        ret = i == 0 ? 1 : 2; /* request renew if the key is not the newest one */
+        /* Request renewal if there is a newer active key available */
+        if (i != 0 && find_ticket_for_encryption(&session_tickets.tickets, time(NULL)) != ticket)
```

Comment:
Can do something like `i != 0 && session_tickets.tickets.entries[i - 1].not_before <= ...)` to do the necessary check, instead of calling find_ticket_for_encryption (that invokes a loop)?

Reply:
Yes, that would work to indicate that there's at least one more recent key that _could_ be valid, although the loop should be pretty fast and exit early. We could also directly check whether `entries[0]` is valid, with the reasoning that we want to issue a new ticket using only the most recent key. I favored calling `find_ticket_for_encryption` because it keeps the whole validity check logic in one place, but other than that, I don't have a strong opinion.